Composer/packages/client/src/components/Adapters/TeamsManifestGenerator.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { v4 as uuidv4 } from 'uuid';
+import { jsx } from '@emotion/core';
+import { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import formatMessage from 'format-message';
+import { Text } from 'office-ui-fabric-react/lib/Text';
+import { IButtonStyles, IconButton } from 'office-ui-fabric-react/lib/components/Button';
+import { FontSizes, NeutralColors } from '@uifabric/fluent-theme/lib/fluent';
+import { useRef } from 'react';
+import { ITextField, TextField } from 'office-ui-fabric-react/lib/components/TextField';
+import { DialogTypes, DialogWrapper } from '@bfc/ui-shared/lib/components/DialogWrapper';
+import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
+
+import { defaultTeamsManifest } from '../../constants';
+
+const iconButtonStyle: IButtonStyles = {
+  root: {
+    height: 'unset',
+    float: 'right',
+    marginRight: '10px',
+  },
+  menuIcon: {
+    backgroundColor: NeutralColors.white,
+    color: NeutralColors.gray130,
+    fontSize: FontSizes.size16,
+  },
+  rootDisabled: {
+    backgroundColor: NeutralColors.white,
+  },
+  rootHovered: {
+    backgroundColor: 'unset',
+  },
+  rootPressed: {
+    backgroundColor: 'unset',
+  },
+};
+
+type TeamsManifestGeneratorProps = {
+  hidden: boolean;
+  botAppId?: string;
+  botDisplayName?: string;
+  onDismiss: () => void;
+};
+
+export const TeamsManifestGenerator = (props: TeamsManifestGeneratorProps) => {
+  const textFieldRef = useRef<ITextField>(null);
+
+  const copyCodeToClipboard = () => {
+    try {
+      textFieldRef.current?.select;
+      textFieldRef.current?.select();
+      document.execCommand('copy');
+      textFieldRef.current?.setSelectionRange(0, 0);
+      textFieldRef.current?.blur();
+    } catch (e) {
+      console.error('Something went wrong when trying to copy manifest content to clipboard.', e);
```

Comment:
we dont do console log anywhere

Reply:
I see quite a few instances of console.error in the client workspace but can change this to fit whatever the standard should be. What do you suggest?

Reply:
Okay I have spoken with Srinaath and implemented the correct front end error handling